module/VuFind/src/VuFind/Search/History.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace VuFind\Search;
+
+use minSO;
+
+class History
+{
+
+    /**
+     * @var \VuFind\Db\Table\Search
+     */
+    protected $searchTable;
+
+    /**
+     * @var \Zend\Session\SessionManager
+     */
+    protected $sessionManager;
+
+    /**
+     * @var \VuFind\Search\Results\PluginManager
+     */
+    protected $resultsManager;
+
+    /**
+     * @var \VuFind\Search\Memory
+     */
+    protected $searchMemory;
+
+    /**
+     * History constructor
+     * @param \VuFind\Db\Table\Search $searchTable
+     * @param \Zend\Session\SessionManager $sessionManager
+     * @param \VuFind\Search\Results\PluginManager $resultsManager
+     * @param \VuFind\Search\Memory $searchMemory
+     */
+    public function __construct($searchTable, $sessionManager, $resultsManager, $searchMemory)
+    {
+        $this->searchTable = $searchTable;
+        $this->sessionManager = $sessionManager;
+        $this->resultsManager = $resultsManager;
+        $this->searchMemory = $searchMemory;
+    }
+
+    /**
+     * @param int $userId
+     * @return array
+     */
+    public function getSearchHistory($userId = null, $purged = false)
+    {
+        // Retrieve search history
+        $searchHistory = $this->searchTable->getSearches(
+            $this->sessionManager->getId(),
```

Comment:
It seems like it may be overkill to inject the session manager in order to extract the session ID here. I would suggest either injecting the session ID through the constructor (since it's essentially a constant value in the context of a single script execution) or else make it a parameter to getSearchHistory (since it's part of the input required to figure out which history to retrieve).